modules/disk-uploader/pkg/vmexport/vmexport.go

```diff
@@ -99,6 +99,28 @@ func GetRawDiskUrlFromVolumes(client kubecli.KubevirtClient, namespace, name, vo
 	return "", fmt.Errorf("volume %s is not found in VirtualMachineExport internal volumes", volumeName)
 }
 
+func GetLabelsFromExportSource(k8sClient *k8sv1.CoreV1Client, virtClient kubecli.KubevirtClient, exportSourceKind, exportSourceNamespace, exportSourceName, volumeName string) (map[string]string, error) {
+	var sourceLabels map[string]string
+	switch exportSourceKind {
+	case "VirtualMachine", "VirtualMachineSnapshot":
+		dv, err := virtClient.CdiClient().CdiV1beta1().DataVolumes(exportSourceNamespace).Get(context.Background(), volumeName, metav1.GetOptions{})
```

Comment:
Can you explain why VM/VMS equals DV? What if the volume of a VM is a PVC without DV?

Reply:
From what I see, `datavolumedisk` is what set in the VM snapshot - From the [example-vm](https://github.com/kubevirt/kubevirt-tekton-tasks/blob/main/examples/pipelines/disk-uploader/disk-uploader-pipeline.yaml#L31-L77) volumes in case you apply it:
```
...
volumes:
   - dataVolume:
     name: example-dv
   name: datavolumedisk
```
I thought that in both kinds, if you specify `example-dv` volume name it'll work. I'll need to check it and test it (haven't tested it yet, because wanted you to review the IT/Pref labels).

Reply:
> I thought that in both kinds, if you specify example-dv volume name it'll work.

I can see that if you apply VMExport on both VM and VMSnapshot, it provides the same volume name (`example-dv` as I mentioned). So this case: `case "VirtualMachine", "VirtualMachineSnapshot"` is valid.

> What if the volume of a VM is a PVC without DV?

In that case, there is a single volume that is name is the PVC name.

Reply:
I've added here [revision](https://github.com/kubevirt/kubevirt-tekton-tasks/compare/225f69c323b79748361dd466cb4092de355fb13f..89ce942291805fc4b6366735db7f64d641269b2d) to test the `GetLabelsFromExportSource()` function from `vmexport.go`.

Reply:
I'd suggest to use an if statement for the VM/Snapshot case and let it fall through to PVC handling, if no DV was found on the VM/Snapshot, instead of duplicating the PVC handling code.

Reply:
duplicating the PVC call is not an option, because you need to provide different names, in case of VM or VMSnapshot kinds, PVC name equals to volume name:
```
virtClient.CoreV1().PersistentVolumeClaims(exportSourceNamespace).Get(context.Background(), volumeName, metav1.GetOptions{})
```
And in case of PVC kind, then just the PVC name:
```
virtClient.CoreV1().PersistentVolumeClaims(exportSourceNamespace).Get(context.Background(), exportSourceName, metav1.GetOptions{})
```

Reply:
Also IMO, switch cases are more testable, clearer, and readable. It makes sense to think of it like if VM or VMSnapshot is not found, then try to get it from PVC directly.

Reply:
You've added a lot of nesting into the switch case, I don't think that makes it more readable. If you want to keep it separate I'd suggest to add two funcs for the different cases and to call them from the switch cases.

Reply:
Done.